clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3501,9 +3501,18 @@ bool Sema::checkTargetAttr(SourceLocation LiteralLoc, StringRef AttrStr) {
   return false;
 }
 
+static bool isArmStreaming(const FunctionDecl *FD) {
+  if (FD->hasAttr<ArmLocallyStreamingAttr>())
+    return true;
```

Comment:
Is this check required? I thought it was more about the interface that can't be both "streaming" and "non-streaming" (for different versions of the function), because the caller needs to know in which mode to call the function. However, the callee is free to change streaming mode in its implementation.

Reply:
Oh, you're right, we don't need this one, since the ABI isn't different from a normal function: https://arm-software.github.io/acle/main/acle.html#changing-streaming-mode-locally